compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -61,6 +62,17 @@ struct WgpDetails {
 struct ChipDetails {
   uint32_t wgpCount;
   std::optional<StringRef> sku;
+  // Aggregate chip-level bandwidth in TB/s.
+  std::optional<float> peakMemoryBandwidthTBs;
+  // Optional per-data-type compute performance (TFLOPs/s).
+  llvm::SmallDenseMap<ComputeBitwidths, float> peakPerfTFLOPs;
+
+  ChipDetails(
+      uint32_t wgp, std::optional<llvm::StringRef> s = std::nullopt,
+      std::optional<float> bw = std::nullopt,
+      std::initializer_list<std::pair<ComputeBitwidths, float>> perf = {})
+      : wgpCount(wgp), sku(s), peakMemoryBandwidthTBs(bw),
+        peakPerfTFLOPs(perf.begin(), perf.end()) {}
```

Comment:
```suggestion
        peakPerfTFLOPs(perf) {}
```

I think this constructor should work: https://github.com/llvm/llvm-project/blob/1633e0ba8b5d550afacd8d21ab179a2e5d5d48ee/llvm/include/llvm/ADT/DenseMap.h#L819

Reply:
This is where I originally hit a compile failure that it can't compile because the argument is no match to `std::initializer_list<typename BaseT::value_type> Vals`. I pondered on it for a second and then decided to use an explicit range to fix it.

Now that you ask, it help me to push a step further. As it turns out my `std::pair<ComputeBitwidths, float>` is the one that corrupted it. Once I change the base type to `llvm::detail::DenseMapPair<ComputeBitwidths, float>` everything works as it should

Reply:
Good find, I think we should relax this a bit and allow other pair types

Reply:
Yah, especially considering that DenseMapPair inherits from std::pair, they are almost similar (but templated types requires exact match).